underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -341,17 +351,18 @@ public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
    */
   private boolean mkdirsInternal(String path) {
     try {
-      String keyAsFolder = addFolderSuffixIfNotPresent(
-          CommonUtils.stripPrefixIfPresent(path, Constants.HEADER_SWIFT));
       // We do not check if a file with same name exists, i.e. a file with name
       // 'swift://swift-container/path' and a folder with name 'swift://swift-container/path/'
       // may both exist simultaneously
-      SwiftDirectClient.put(mAccess, keyAsFolder).close();
-      return true;
+      final OutputStream out = createOutputStream(addFolderSuffixIfNotPresent(path));
+      if (out != null) {
+        out.close();
+        return true;
+      }
```

Comment:
This pattern seems to be inconsistent with the one above. Should `createOutputStream` guarantee non-null return value?